hibernate-core/src/test/java/org/hibernate/orm/test/embeddable/JdbcTypeCodeJsonWithTableSchemaNameTest.java

```diff
@@ -0,0 +1,137 @@
+package org.hibernate.orm.test.embeddable;
+
+import java.util.List;
+import java.util.Map;
+
+import org.hibernate.annotations.JdbcTypeCode;
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.dialect.PostgreSQLDialect;
+import org.hibernate.orm.test.jpa.BaseEntityManagerFunctionalTestCase;
+import org.hibernate.type.SqlTypes;
+
+import org.hibernate.testing.orm.junit.JiraKey;
+import org.hibernate.testing.orm.junit.RequiresDialect;
+import org.junit.Test;
+
+import jakarta.persistence.Access;
+import jakarta.persistence.AccessType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Embeddable;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import jakarta.persistence.Tuple;
+import jakarta.transaction.Transactional;
+
+import static org.hibernate.testing.transaction.TransactionUtil.doInJPA;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@RequiresDialect(PostgreSQLDialect.class)
+@JiraKey( "HHH-16612" )
+public class JdbcTypeCodeJsonWithTableSchemaNameTest extends BaseEntityManagerFunctionalTestCase {
```

Comment:
It would be nice if you could use the annotation test class style (`@SessionFactory`, `@ServiceRegistry` and `@DomainModel`) instead of extending the old base class. You also don't need the `@Transactional` annotation on test methods.

Reply:
Changed